pkg/clients/database/mysql_test.go

```diff
@@ -313,3 +321,93 @@ func TestUpdateMySQLVirtualNetworkRuleStatusFromAzure(t *testing.T) {
 		})
 	}
 }
+
+func TestFetchAsyncOperation(t *testing.T) {
+	inprogressStatus := "inprogress"
+	inProgressResponse := fmt.Sprintf(`{"status": "%s"}`, inprogressStatus)
+
+	errorStatus := "Failed"
+	errorResponse := fmt.Sprintf(`{"status": "%s"}`, errorStatus)
+	errorMessage := fmt.Sprintf(`Code="Failed" Message="The async operation failed." AdditionalInfo=[{"status":"%s"}]`, errorStatus)
+
+	pollingURL := "https://crossplane.io"
+
+	type args struct {
+		sender autorest.Sender
+		as     *v1beta1.AsyncOperation
+	}
+	type want struct {
+		op  *v1beta1.AsyncOperation
+		err error
+	}
+	cases := map[string]struct {
+		args args
+		want want
+	}{
+		"Should skip when there is no operation": {},
```

Comment:
Definitely open to push back here, but would you be opposed to actually passing in `nil` operation here? I know this is what is happening with the zero value, but it feels like it might be a little more readable to me :)

Reply:
Yeah, I feel like @negz will jump and say _hey! this is already `nil`_ 😄

Reply:
We'll defer to @negz on this then :)